userland/Makefile

```diff
@@ -31,6 +31,9 @@ CPPFLAGS += \
 	    -mpic-register=r9\
 	    -mno-pic-data-is-text-relative
 
+$(BUILDDIR)/%.o: %.c | $(BUILDDIR)
+	$(CC) $(CFLAGS) $(CPPFLAGS) -c -o $@ $<
+
```

Comment:
I'm not sold that this is the right choice. That feels like a lot of copy-pasting for a really weird corner case (what app actually needs a different compiler?). It's worth noting that we have some compilation expectations (i.e. `pic`) that people would need to get correct if they rolled their own.

I don't quite understand your ble library example here. The ble-serialization app currently builds with the common ruleset and is actually a great example of how to extend things.

Finally, if people really need to do something different / fancy, they can follow the `libtock` example, just make their own build directory for their fancy stuff and the `$(BUILDDIR)/%.o` pattern won't match. That feels like a fine escape hatch for the esoteric case while making the common case easy.